front/src/modules/pathfinding/components/Itinerary/ModalSuggestedVias.tsx

```diff
@@ -32,7 +35,10 @@ const ModalSuggestedVias = ({ suggestedVias }: ModalSuggestedViasProps) => {
     [destination]
   );
 
-  const removeViaFromPath = (op: SuggestedOP) => dispatch(removeVia(op));
+  const removeViaFromPath = (op: SuggestedOP) => {
+    const newPathSteps = pathSteps.filter((step) => !step || matchPathStepAndOp(step, op));
```

Comment:
I don't see how this could have ever work. Here we filter the steps which matches the op we want to remove so we will have only the removed op in path steps at the end.

Also, same as the other comment, here we know we won't have any null `pathSteps` so we should be able to remove the `!step` (if you did the suggested changed at the top of the file)

```suggestion
    const newPathSteps = pathSteps.filter((step) => !matchPathStepAndOp(step, op));
```

Reply:
Done !